### ISSUE TITLE
Putting graphviz href attr in output of dataflow to alllow hyperlinks in svg

### ISSUE THREAD

**Issue by tjtaill** [sonyxperiadev/dataflow#6]
Hi I really enjoy using your dataflow tool so much better than visio. I was wondering how hard it would be to modify your dataflow code so that I could put an href attribute in the generated graphviz dot output. This would allow the creationm of svg's  that embeds hyperlinks. Ideally I would like this to be doable by adding an attribute in your dataflow language ?

function f {
  title = "f"
  href = "1"
}

that would create a node in the dot like

f [
 label = "f";
 href = "1";
]

I am not a haskell expert but I am hoping that it will be relatively easy

**Comment by owickstrom** [comment]
Thank you, glad you like it!

About your feature proposal, I think it's a good idea. It's been a while since I did something in this project, but I'm pretty sure it's an easy addition. Do you want to have a try at it? In that case you should tweak the relevant patterns in `convertNode` (https://github.com/sonyxperiadev/dataflow/blob/master/src/DataFlow/DFD.hs#L56), and preferably add a helper in [`Attributes.hs`](https://github.com/sonyxperiadev/dataflow/blob/master/src/DataFlow/Attributes.hs) that takes the Attributes map, and a string, and returns a string with the link markup if the `href` attribute is present. 

IIRC you also need to add a namespace to the SVG XML root element to make links work, not sure how that can be done with graphviz.

If you don't have time, or don't feel confident in trying to implement it, I can probably find some time to do it this weekend.

**Comment by tjtaill** [comment]
Oskar,

I managed to get it to work yesterday.
The links work graphviz properly creates them in the svg I have used
graphviz in the past to do this.
It is working great thanks.

Troy

**Comment by owickstrom** [comment]
OK, without any modification to DataFlow? Would you mind sharing a small example? :)

**Comment by tjtaill** [comment]
Oskar,

Sorry for the long delay work is hectic. I did modify the dataflow source
code and here is the example. Basically I use jedit http://www.jedit.org/
and batik https://xmlgraphics.apache.org/batik/

To create an interactive editor for drawing languages like plantuml, dot,
tikz and now dataflow. I encode the line numbers as svg links and when I
click on them in the  batik svg viewer it jumps to the line in
source file in jedit. I even created syntax highlighting for you language
and code snippets in jedit so it is very quick and enjoyable to create
those data flow diagrams.

I notice you produce plantuml sequence diagrams too. I imagine with some
work I could get the svg hyperlink line number references to work their too.

I really hate visio so I am glad I came up across your solution thanks again

Troy
